src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -744,6 +813,47 @@ Error procSetTitle(const json::JsonRpcRequest& request,
    return Success();
 }
 
+Error procEraseBuffer(const json::JsonRpcRequest& request,
+                      json::JsonRpcResponse* pResponse)
+{
+   std::string handle;
+
+   Error error = json::readParams(request.params,
+                                  &handle);
+   if (error)
+      return error;
+
+   ProcTable::const_iterator pos = s_procs.find(handle);
+   if (pos == s_procs.end())
+   {
+      return systemError(boost::system::errc::invalid_argument, ERROR_LOCATION);
+   }
+
+   pos->second->deleteLogFile();
+   return Success();
+}
+
+Error procGetBuffer(const json::JsonRpcRequest& request,
+                    json::JsonRpcResponse* pResponse)
+{
+   std::string handle;
+
+   Error error = json::readParams(request.params,
+                                  &handle);
+   if (error)
+      return error;
+
+   ProcTable::const_iterator pos = s_procs.find(handle);
+   if (pos == s_procs.end())
+   {
+      return systemError(boost::system::errc::invalid_argument, ERROR_LOCATION);
+   }
+
+   pResponse->setResult(pos->second->getSavedBuffer());
```

Comment:
Probably already on your TODO but we'll need to throttle this at some point -- it's easy to run a terminal command which generates huge amounts of output and we don't want that to overwhelm the client